### PR TITLE
Translation's language should match object's language when saving translations group

### PR DESCRIPTION
**Pull request by raaaahman**
## Before

There is no verification of the object language when it is assigned to a translation group. This can lead to translations group having several time the same object with different languages like: 

```php
$polylang->model->post->save_translations( 1, array( 'en' => 1, 'fr' => 1, 'es' => 1 );
```

Although, this was verified when saving a post or media from the classic editor in `PLL_Admin_Filters_Post` and `PLL_Admin_Filters_Media`.

## Changes

Move the protection against incorrectly constituted translations group in the `PLL_Translated_Object::save_translations()` method to improve robustness.

## Related issues 

- https://github.com/polylang/polylang-pro/issues/808
- https://github.com/polylang/polylang-pro/pull/810

### CODE REVIEW

**Review by Marianne380 — APPROVED** [review]

File: include/translated-object.php [hunk 1]
```diff
@@ -188,13 +188,16 @@ protected function should_update_translation_group( $id, $translations ) {
 	 * @since 0.5
 	 *
 	 * @param int   $id           Object id ( typically a post_id or term_id ).
-	 * @param int[] $translations An associative array of translations with language code as key and translation id as value.
-	 * @return void
+	 * @param array $translations An associative array of translations with language code as key and translation id as value.
```
Comment (Chouby) [inline include/translated-object.php:191]: Why did you change this doc line?

File: include/translated-object.php [hunk 2]
```diff
@@ -236,9 +239,29 @@ public function save_translations( $id, $translations ) {
...
+			return array();
 		}
 	}
 
+	/**
+	 * Returns translations after checking the translated post is in the right language
+	 *
+	 * @param array $translations Locales as keys and post ids as values.
+	 *
+	 * @return array
```
Comment (Chouby) [inline include/translated-object.php:253]: Could you add `@since` and precise the array to `int[]` ? Also the keys are not locales but language codes.

File: include/translated-object.php (hunk 1 above)
Comment (raaaahman) [reply include/translated-object.php]: I did a mistake while resolving a merge conflict.
